main.go

```diff
@@ -275,11 +277,26 @@ func init() {
 
 	cmd = kingpin.MustParse(cli.Parse(os.Args[1:]))
 
+	// Configure logging.
 	switch {
 	case *trace:
 		log.SetLevel(5)
 	case *debug:
 		log.SetLevel(2)
+	default:
+		l := int8(*logLevel)
+		if l < -1 || l > 5 {
+			fmt.Fprintf(os.Stderr, "invalid log level: %d\n", *logLevel)
+			os.Exit(1)
+		}
+
+		if l == -1 {
+			// Zap uses "5" as the value for fatal.
+			// We need to pass in "-5" because `SetLevel` passes the negation.
```

Comment:
I have some dumb questions:
* What does "uses 5 as the value for fatal" mean here? Are fatal errors are logged at level 5?
* Why does `SetLevel` pass the negation? That seems really strange.

Reply:
> What does "uses 5 as the value for fatal" mean here? Are fatal errors are logged at level 5?

[Zap's logging levels](https://github.com/uber-go/zap/blob/0ba452dbe15478739ad4bab1067706018a3062c6/zapcore/level.go#L34) use `0` for info, with negatives increasing verbosity and 1-5 decreasing verbosity.

TruffleHog only seems to use levels <= 0, so setting the level to `5` effectively prevents anything from being logged.

> Why does SetLevel pass the negation? That seems really strange.

My guess is that someone preferred the look of (+)1->5 for verbosity and not (-)1->5. 🤷‍♂️

Reply:
It originates from the combination of `zap` as the log implementation for `logr`.

* `logr` defines higher numbers as being more verbose
* `zap` defines lower numbers as being more verbose

I left a comment about it in `SetLevel`: https://github.com/trufflesecurity/trufflehog/blob/2a010914749d701e8921f3291fb808960cae989a/pkg/log/level.go#L27-L29

Reply:
Ah, I didn't realize it was two different implementations.